great_expectations/core/expectation_suite.py

```diff
@@ -176,6 +181,19 @@ def __repr__(self):
     def __str__(self):
         return json.dumps(self.to_json_dict(), indent=2)
 
+    def __deepcopy__(self, memo: dict):
+        cls = self.__class__
+        result = cls.__new__(cls)
+
+        memo[id(self)] = result
+
+        attributes_to_copy = list(ExpectationSuiteSchema().fields.keys())
+        for key in attributes_to_copy:
+            setattr(result, key, deepcopy(getattr(self, key)))
+
+        setattr(result, "_data_context", self._data_context)
+        return result
+
```

Comment:
`__deepcopy__` has been over-written because we don't want to copy `DataContext` reference, which will cause a recursion error.